.github/workflows/test.yml

```diff
@@ -25,22 +25,18 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        # maximize CI coverage of different platforms and python versions while minimizing the
+        # total number of jobs. We run all pytest splits with the oldest supported python
+        # version (currently 3.9) on windows (seems most likely to surface errors) and with
+        # newest version (currently 3.12) on ubuntu (to get complete coverage on unix). We
+        # ignore mac-os, which is assumed to be similar to ubuntu.
+        config:
+          - { os: windows-latest, python: "3.9", resolution: highest }
+          - { os: ubuntu-latest, python: "3.12", resolution: lowest-direct }
```

Comment:
Is this a typo? Why "Python 3.12" is the "lowest-direct" while "3.9" is the "highest"?